pkg/asset/manifests/gcp/cluster.go

```diff
@@ -0,0 +1,171 @@
+	"github.com/openshift/installer/pkg/asset/manifests/capiutils"
+	"github.com/openshift/installer/pkg/types/gcp"
+)
+
+// GenerateClusterAssets generates the manifests for the cluster-api.
+func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID *installconfig.ClusterID) (*capiutils.GenerateClusterAssetsOutput, error) {
+	manifests := []*asset.RuntimeFile{}
+
+	const (
+		description = "Created By OpenShift Installer"
+	)
+
+	networkName := fmt.Sprintf("%s-network", clusterID.InfraID)
+	if installConfig.Config.GCP.Network != "" {
+		networkName = installConfig.Config.GCP.Network
+	}
+
+	masterSubnet := gcp.DefaultSubnetName(clusterID.InfraID, "master")
+	if installConfig.Config.GCP.ControlPlaneSubnet != "" {
+		masterSubnet = installConfig.Config.GCP.ControlPlaneSubnet
+	}
+
+	master := capg.SubnetSpec{
+		Name:        masterSubnet,
+		CidrBlock:   "",
+		Description: ptr.To(description),
+		Region:      installConfig.Config.GCP.Region,
+	}
+
+	workerSubnet := gcp.DefaultSubnetName(clusterID.InfraID, "worker")
+	if installConfig.Config.GCP.ComputeSubnet != "" {
+		workerSubnet = installConfig.Config.GCP.ComputeSubnet
+	}
+
+	worker := capg.SubnetSpec{
+		Name:        workerSubnet,
+		CidrBlock:   "",
+		Description: ptr.To(description),
+		Region:      installConfig.Config.GCP.Region,
+	}
+
+	// Add the CIDR information.
+	machineV4CIDRs := []string{}
+	for _, network := range installConfig.Config.Networking.MachineNetwork {
+		if network.CIDR.IPNet.IP.To4() != nil {
+			machineV4CIDRs = append(machineV4CIDRs, network.CIDR.IPNet.String())
+		}
+	}
+
+	if len(machineV4CIDRs) == 0 {
+		return nil, fmt.Errorf("failed to parse machine CIDRs")
+	}
+
+	_, ipv4Net, err := net.ParseCIDR(machineV4CIDRs[0])
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse machine network CIDR: %w", err)
+	}
+
+	if installConfig.Config.GCP.ControlPlaneSubnet == "" {
+		masterCIDR, err := cidr.Subnet(ipv4Net, 1, 0)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create the master subnet %w", err)
+		}
+		master.CidrBlock = masterCIDR.String()
+	}
+
+	if installConfig.Config.GCP.ComputeSubnet == "" {
+		workerCIDR, err := cidr.Subnet(ipv4Net, 1, 1)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create the worker subnet %w", err)
+		}
+		worker.CidrBlock = workerCIDR.String()
+	}
+
+	subnets := []capg.SubnetSpec{master, worker}
+
+	labels := map[string]string{}
+	labels[fmt.Sprintf("kubernetes-io-cluster-%s", clusterID.InfraID)] = "owned"
+	labels[fmt.Sprintf("capg-cluster-%s", clusterID.InfraID)] = "owned"
+	for _, label := range installConfig.Config.GCP.UserLabels {
+		labels[label.Key] = label.Value
+	}
+
+	// Find the availability zones and use this information to configure the GCP Cluster Failure Domains.
+	zones := sets.New[string]()
+	defaultMachinePlatformDefined := false
+	if installConfig.Config.Platform.GCP != nil {
+		if installConfig.Config.Platform.GCP.DefaultMachinePlatform != nil {
```

Comment:
nit, you can condense this a bit like
```suggestion
	if gcp := installConfig.Config.Platform.GCP; gcp != nil && gcp.DefaultMachinePlatform != nil {
```